keras_cv/layers/preprocessing3d/random_pasting_objects.py

```diff
@@ -0,0 +1,242 @@
+    """A preprocessing layer which randomly pastes object point clouds and bounding boxes during training.
+
+    This layer will randomly paste object point clouds and bounding boxes.
+    During inference time, the output will be identical to input. Call the layer with `training=True` to paste bounding boxes.
+
+    Input shape:
+      point_clouds: 3D (multi frames) float32 Tensor with shape
+        [num of frames, num of points, num of point features].
+        The first 5 features are [x, y, z, class, range].
+      bounding_boxes: 3D (multi frames) float32 Tensor with shape
+        [num of frames, num of boxes, num of box features].
+        The first 8 features are [x, y, z, dx, dy, dz, phi, box class].
+
+    Output shape:
+      A tuple of two Tensors (point_clouds, bounding_boxes) with the same shape as input Tensors.
+
+    Arguments:
+      label_index: An optional int scalar sets the target object index.
+        Bounding boxes and corresponding point clouds with box class == label_index will be saved as OBJECT_BOUNDING_BOXES and OBJECT_POINT_CLOUDS.
+        If label index is None, all valid bounding boxes (box class !=0) are used.
+      min_paste_bounding_boxes: A int scalar sets the min number of pasted bounding boxes.
+      max_paste_bounding_boxes: A int scalar sets the max number of pasted bounding boxes.
+      
+    """
+
+    def __init__(
+        self, label_index=None, min_paste_bounding_boxes=0, max_paste_bounding_boxes=10, **kwargs
+    ):
+        super().__init__(**kwargs)
+        if label_index and label_index < 0:
+            raise ValueError("label_index must be >=0.")
+        if min_paste_bounding_boxes < 0:
+            raise ValueError("min_paste_bounding_boxes must be >=0.")
+        if max_paste_bounding_boxes < 0:
+            raise ValueError("max_paste_bounding_boxes must be >=0.")
+        if max_paste_bounding_boxes < min_paste_bounding_boxes:
+            raise ValueError(
+                "max_paste_bounding_boxes must be >= min_paste_bounding_boxes."
+            )
+
+        self._label_index = label_index
+        self._min_paste_bounding_boxes = min_paste_bounding_boxes
+        self._max_paste_bounding_boxes = max_paste_bounding_boxes
+        self._iou_3d = iou_3d.IoU3D()
+
+    def get_random_transformation(
+        self,
+        point_clouds,
+        bounding_boxes,
+        object_point_clouds,
+        object_bounding_boxes,
+        **kwargs
+    ):
+        del point_clouds
+        num_paste_bounding_boxes = self._random_generator.random_uniform(
+            (),
+            minval=self._min_paste_bounding_boxes,
+            maxval=self._max_paste_bounding_boxes,
+        )
+        num_paste_bounding_boxes = tf.cast(num_paste_bounding_boxes, dtype=tf.int32)
+        num_existing_bounding_boxes = tf.shape(bounding_boxes)[1]
+        if self._label_index:
+            object_mask = tf.math.equal(
+                object_bounding_boxes[0, :, 7], self._label_index
```

Comment:
nit: should we consider defining this magic number?
BOX_LABEL_INDEX = 7

Reply:
Done.